hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -354,10 +365,8 @@ protected void onRemoveAsyncInternal(Data keyData, ClientDelegatingFuture future
                 completionId);
         ClientInvocationFuture future = invoke(request, keyData, completionId);
         ClientDelegatingFuture<T> delegatingFuture = newDelegatingFuture(future, GET_AND_REPLACE_RESPONSE_DECODER);
-        if (async) {
-            ExecutionCallback callback = statisticsEnabled ? new OnReplaceCallback(startNanos, true) : null;
-            onReplaceAndGetAsync(keyData, newValueData, newValue, delegatingFuture, callback);
-        }
+        ExecutionCallback callback = async && statisticsEnabled ? new OnReplaceCallback(startNanos, true) : null;
+        onReplaceAndGetAsync(keyData, newValueData, newValue, delegatingFuture, callback);
```

Comment:
We always have to call `onReplaceAndGetAsync()`, otherwise the Near Cache proxy cannot add it's invalidation callback wrapper. All methods can handle a `null` callback fine, so it's just created if `async` and `statisticsEnabled` are both `true`.